Plugins/Creature/Extras/BonusStacking.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "API/Types.hpp"
+#include "Common.hpp"
+#include "Services/Hooks/Hooks.hpp"
+#include "Services/Events/Events.hpp"
+
+class CNWSCreature;
+class CNWSObject;
+
+namespace Creature {
+
+namespace NostackMode
+{
+typedef enum
+{
+	DISABLED,
+	NO_STACKING,
+	ALLOW_ONE_SPELL,
+	ALLOW_ALL_SPELLS,
+	CUSTOM_TYPES
+} TYPE;
+}
+
+namespace NostackType
+{
+typedef enum
+{
+	ENHANCEMENT,
+	CIRCUMSTANCE,
+	COMPETENCE,
+	INSIGHT,
+	LUCK,
+	MORALE,
+	PROFANE,
+	RESISTANCE,
+	SACRED,
+	CUSTOM,
```

Comment:
Why not say everything <100 is reserved, and everything >=100 is custom, rather than having explicit custom values.

Reply:
Removed the custom tags and limited the maximum value to 20, allowing for 12 (9-20) user-defined types.